rotkehlchen/premium/sync.py

```diff
@@ -254,8 +256,10 @@ def _abort_new_syncing_premium_user(
         premium API keys and we failed. But a directory was created. Remove it.
         But create a backup of it in case something went really wrong
         and the directory contained data we did not want to lose"""
+        user_data_dir = self.data.user_data_dir
+        self.data.db.logout()  # wipes self.data.user_data_dir
```

Comment:
Hmmm it already seems to happen here. So perhaps with this you don't need the explicit disconnects above?
```suggestion
        self.data.db.logout()  # wipes self.data.user_data_dir, so store it
```

Reply:
Yes you're correct. Doing this fixes one of the tests, but unfortunately doesn't fix the other. I must have ran the same test twice locally before pushing thinking it was the two failing ones. I am going to have to re-think this one as it's not as simple as I first thought. I am thinking of removing `_abort_new_syncing_premium_user` altogether...

This is actually a genuine bug in production though. I was able to reproduce it and got the same error:
<img width="286" alt="image" src="https://github.com/rotki/rotki/assets/28752776/5ee66fd4-ca8a-4d84-8483-0fd4b813fed3">